### PR TITLE
add chatter detection

### PR DESCRIPTION
**Pull request by zekth**
Some folks asked for chatter detection in the keyboard tester. So here it is:
Screenshot 2019-08-13 at 10 42 12

### CODE REVIEW

File: src/components/VisualTesterKeymap.vue [hunk 1]
```diff
@@ -208,10 +250,10 @@ export default {
       }
       this.scrollToEnd();
     },
-    formatKeyEvent(ev, endTS) {
+    formatKeyEvent(ev, time) {
       let msg = [];
-      if (endTS) {
-        msg.push(`in ${(endTS - this.timing[ev.code]).toFixed(3)}ms`);
```
Comment (yanfali) [inline src/components/VisualTesterKeymap.vue]: why did you remove the precision limit?

File: src/components/VisualTesterKeymap.vue [hunk 2]
```diff
@@ -161,9 +187,22 @@ export default {
         evStr
       ].join(' ');
     },
+    getElapsedTime(ev, endTs) {
+      return (endTs - this.timing[ev.code]).toFixed(3);
+    },
+    enableCatchKeys() {
```
Comment (yanfali) [inline src/components/VisualTesterKeymap.vue]: I don't really like this naming. I think it should be something like enableInput and IgnoreInput

File: src/components/VisualTesterKeymap.vue (hunk 1 above)
Comment (zekth) [reply src/components/VisualTesterKeymap.vue]: it has just been moved into a function: https://github.com/qmk/qmk_configurator/pull/466/files#diff-44b9ededf7579010954e5efecdb111eeR188

File: src/components/VisualTesterKeymap.vue (hunk 2 above)
Comment (zekth) [reply src/components/VisualTesterKeymap.vue]: refactored

File: src/components/VisualTesterKeymap.vue [hunk 3]
```diff
@@ -161,9 +185,21 @@ export default {
         evStr
       ].join(' ');
     },
+    getElapsedTime(ev, endTs) {
+      return (endTs - this.timing[ev.code]).toFixed(3);
+    },
+    createKeyListenners() {
```
Comment (yanfali) [inline src/components/VisualTesterKeymap.vue]: Listeners

File: src/components/VisualTesterKeymap.vue [hunk 4]
```diff
@@ -161,9 +185,21 @@ export default {
...
+      document.addEventListener('keydown', this.keydown);
+      document.addEventListener('keyup', this.keyup);
+    },
+    destroyKeyListenners() {
```
Comment (yanfali) [inline src/components/VisualTesterKeymap.vue]: Ditto

File: src/components/VisualTesterKeymap.vue [hunk 5]
```diff
@@ -50,6 +50,22 @@
         spellcheck="false"
         v-html="status"
       ></div>
+      <div id="chatter-container">
+        <label>Chatter Threshold:</label>
```
Comment (yanfali) [inline src/components/VisualTesterKeymap.vue]: i18n. Can you add a Unit of Measurement in (Milliseconds)

File: src/components/VisualTesterKeymap.vue [hunk 6]
```diff
@@ -50,6 +50,22 @@
...
+          @focus="destroyKeyListenners"
+          @blur="createKeyListenners"
+          type="number"
+          v-model="chatterThreshold"
+          min="1"
+          max="100"
+          step="1"
+        />
+        <span id="chatter-alert" v-show="chatterDetected"
+          >CHATTER HAS BEEN DETECTED!</span
```
Comment (yanfali) [inline src/components/VisualTesterKeymap.vue]: i18n please

File: src/scss/themes.scss [hunk 7]
```diff
@@ -31,6 +31,13 @@ html {
     #app {
       color: #2c3e50;
     }
+    #chatter-threshold {
+      background-color: #EEEEEE;
+      border:none;
+    }
+    .chatter-detected {
+      background-color: rgb(233, 107, 107) !important;
```
Comment (yanfali) [inline src/scss/themes.scss]: why important?

Comment (zekth) [reply src/scss/themes.scss]: can do a bit of refactor on it, but it's due to `detected` class and i'm not sure of how the output of vue is done for the CSS and how the browser will prioritize the stylesheet rules.

Comment (zekth) [reply src/scss/themes.scss]: 2019-08-14 08_37_11-QMK Configurator

got to use important here

**Comment by zekth** [comment]
done the changes.

File: src/scss/themes.scss (hunk 7 above)
Comment (yanfali) [reply src/scss/themes.scss]: @noroadsleft  any suggestions?

Comment (yanfali) [reply src/scss/themes.scss]: usually more specific wins

Comment (zekth) [reply src/scss/themes.scss]: or i can do : 
```
.tester-key.detected.chatter-detected,
.tester-key.chatter-detected {
...
```

Comment (noroadsleft) [reply src/scss/themes.scss]: > or i can do :
> 
> ```
> .tester-key.detected.chatter-detected,
> .tester-key.chatter-detected {
> ...
> ```

I like this better.

Everything else looks good.

**Comment by zekth** [comment]
> Just the one comment regarding the CSS rules, but otherwise I call this good to go.

Done
